util/verible-format.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import os
+import argparse
+import subprocess
+import logging
+import difflib
+
+logger = logging.getLogger('verible_format')
+
+VERIBLE_ARGS = ["--formal_parameters_indentation=indent",
+                "--named_parameter_indentation=indent",
+                "--named_port_indentation=indent",
+                "--port_declarations_indentation=indent"]
+
+
+def get_repo_top():
+    repo_top = subprocess.Popen('git rev-parse --show-toplevel',
+                                stdout=subprocess.PIPE, shell=True)
+    return repo_top.stdout.read().decode().strip()
+
+
+def get_verible_executable_path():
+    proc = subprocess.Popen('which verible-verilog-format',
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE,
+                            shell=True)
+    return proc.stdout.read().decode().strip()
+
+
+def is_verible_formatter_installed():
+    return len(get_verible_executable_path()) > 0
+
+
+def get_verible_version():
+    verible_version = subprocess.Popen('verible-verilog-format --version',
+                                       stdout=subprocess.PIPE,
+                                       stderr=subprocess.PIPE, shell=True).stdout.read()
+    return verible_version.decode().split('\n')[0]
+
+
+def process_file(filename_abs, filename, inplace=False,
+                 show_diff=False, show_cst=False):
+    args = [get_verible_executable_path()] + VERIBLE_ARGS
+    if inplace:
+        args.append('--inplace')
+    args.append(filename_abs)
+
+    verible = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = verible.communicate()
+
+    ret = 0
+
+    if len(stderr) > 0:
+        logger.info('ERROR: ' + str(filename))
+        logger.debug(stderr.decode())
+        ret = 1
+    elif not inplace:
+        with open(filename_abs, 'r') as fp:
+            orig = fp.read().split('\n')
+            formatted = stdout.decode().split('\n')
+
+            diff = []
+            for line in difflib.unified_diff(orig, formatted, lineterm='', n=3,
+                                             fromfile=filename,
+                                             tofile=filename + '.formatted'):
+                diff.append(line)
```

Comment:
Do you actually need this loop? Could you just cast with `diff = list(difflib.unified_diff(...))`?

Reply:
So simple. Love that. Replaced. Thanks!